its/src/test/java/com/sonar/it/scanner/msbuild/utils/BuildCommand.java

```diff
@@ -56,28 +69,65 @@ public BuildCommand addArgument(String... arguments) {
     return this;
   }
 
-  public BuildResult execute(Orchestrator orchestrator) {
-    // ToDo: SCAN4NET-312: Extract createCommand, add support for parameters, add support for MsBuild vs dotnet. Should be similar to ScannerCommand.execute()
-    // ToDo: SCAN4NET-312: Run restore & rebuild. Restore could be via NuGet (on request, or every time for simplicity?), MsBuild or dotnet
-    var environmentVariables = environment.entrySet().stream().map(x -> new EnvironmentVariable(x.getKey(), x.getValue())).toList();
-    var result = dotnetCommand == null
-      ? TestUtils.runMSBuild(orchestrator, projectDir, environmentVariables, timeout, prependArgument("/t:Restore,Rebuild"))
-      : TestUtils.runDotnetCommand(projectDir, environmentVariables, dotnetCommand, dotnetArguments());
+  public BuildResult execute() {
+    addDefaultArguments();
```

Comment:
This will break if you run it twice. And some of our ITs need to run the analysis twice like Incremental PR analysis, or JRE provisioning cache